src/data/redux.test.jsx

```diff
@@ -842,42 +842,10 @@ describe('External API integration tests', () => {
       });
     });
 
-    it('Should dispatch get exam access token', async () => {
+    it('Should return rejected promise no exam', async () => {
       const mockDispatch = jest.fn(() => store.dispatch);
       const mockState = jest.fn(() => store.getState);
-      const dispatchReturn = fetchExamAccess(mockDispatch, mockState);
-      expect(dispatchReturn).toBeInstanceOf(Promise);
-    });
-  });
-});
-
-describe('External API integration tests', () => {
-  let store;
-
-  describe('Test isExam', () => {
-    it('Should return false if exam is not set', async () => {
-      expect(isExam()).toBe(false);
-    });
-  });
-
-  describe('Test getExamAccess', () => {
-    it('Should return empty string if no access token', async () => {
-      expect(getExamAccess()).toBe('');
```

Comment:
It looks like test coverage is now missing for the getExamAccess function, is there a reason these were removed?

Reply:
Ah I see! This was actually a duplicate of the existing tests. But I did modify this specific test to handle the rejected promise case, which doesn't call get exam access. So I just need to add that case back. I think one thing that I was having trouble with in the past is that the mock/test store in the tests is different from the store that is checked in the api functions themselves. I was having trouble mocking that but I will look into mocking that now so that I can still make the call with the test store rather than a newly created store.

Reply:
As I was trying to add back test coverage, I realized that store.getState() is really only used in testing and in these API functions. I believe because typically the state is accessed in context within a component but for these API functions, they are happening outside of the component/thunks. 

In the previous PR, I only tested the case where exam id was null, but in order to have coverage of the exam access thunk, I need to reconcile the issue I was having before. Basically, the test store that is initialized in redux doesn't correspond to the store/state that is accessed/called in the api functions (isExam, getExamAccess, fetchExamAccess). When I tried to test, I wasn't sure how to test because the store that the tests are checking are the test store, while the implementation checks the store that is called. So I'm essentially not sure how to mock this out without mocking the entire implementation which doesn't seem correct.

Any ideas, @Zacharis278 or @alangsto ? Thanks!

Reply:
I don't think there should be much difference from running the thunk tests since we also aren't initializing any components in that case either. Are you using https://github.com/openedx/frontend-lib-special-exams/blob/3239296249ac225a8eeb8a562e7fc1e06eb9f90c/src/setupTest.js#L48 to setup the store?